krun_ext_common.py

```diff
@@ -30,11 +237,82 @@ def run(args, benchmark, num_iters, tmp_file):
         assert headers[1] == "nanos"
 
         for i, row in enumerate(rows):
-            assert row[0] == benchmark
-            wallclock_times[i] = int(row[1]) / 1000000000  # nanos -> seconds.
+            assert row[0] == done_exec.benchmark
+            # nanos -> seconds.
+            wallclock_times[i] = int(row[1]) / 1000000000
+
+    assert len(wallclock_times) == done_exec.num_iters
+    return wallclock_times
+
+
+def post_process_dacapo(done_exec):
+    wallclock_times = [-0.0] * done_exec.num_iters
+    idx = 0
+
+    # Dacapo offers no way to write the results to a file, so we have to scrape
+    # the timings from stderr.
+    for line in done_exec.stderr.splitlines():
+        if not line.startswith("====="):
+            continue
+        if "completed warmup" not in line:
+            continue
+        assert done_exec.benchmark in line
+        line = line.split()
+        index = line.index("in")
+        assert line[index + 2] == "msec"
+        wallclock_times[idx] = float(Decimal(line[index + 1]) / 1000)
+        idx += 1
+
+    return wallclock_times
+
+
+def post_process_spec(done_exec):
+    # We have already check for non-zero exit status, but in addition SPEC
```

Comment:
"check" -> "checked".

Although this whole comment is quite lengthy and hard to parse. Can we tighten it up?

Reply:
sure